AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Dependency/AuthenticationProviderAdapter+DeleteUser.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+#if COCOAPODS
+import AWSMobileClient
+#else
+import AWSMobileClientXCF
+#endif
+
+extension AuthenticationProviderAdapter {
+
+    func deleteUser(request: AuthDeleteUserRequest, completionHandler: @escaping (Result<Void, AuthError>) -> Void) {
+        awsMobileClient.deleteUser(signOut: false) { [weak self] error in
+            guard let error = error else {
+                let signOutOptions = AuthSignOutRequest.Options(globalSignOut: true)
+                let signOutRequest = AuthSignOutRequest(options: signOutOptions)
+                self?.signOut(request: signOutRequest) { result in
+                    switch result {
+                    case .success:
+                        completionHandler(.success(()))
+                        return
+                    case .failure(let error):
+                        completionHandler(.failure(AuthErrorHelper.toAuthError(error)))
+                        return
+                    }
+                }
+                return
```

Comment:
Looks like we wouldn't invoke `completionHandler` if `self` is nil for some reason. That may be expected, but wanted to call it out.

Reply:
Correct. The instance of AuthenticationProvider that is referenced by self is retained during plugin configuration and cannot be nil if the plugin is successfully configured. It the plugin configuration is invalid, configuration will fail and we will never reach this point. An alternative would be to use `guard let self = self else { return }` to make the intent explicit. With a single call, both styles are used elsewhere in the plugin.